src/Core/src/Handlers/Picker/PickerHandler.Android.cs

```diff
@@ -147,6 +148,8 @@ void OnClick(object? sender, EventArgs e)
 				if (_dialog == null)
 					return;
 
+				_dialog.UpdateFlowDirection(VirtualView);
```

Comment:
I don't think you need all these extension methods and calls to GetEffectiveFlowDirection. The PlatformView already tells you what layout direction to use. I'm betting all you need is something like 

```
var dv = _dialog?.Window?.DecorView;
if (dv != null)
{
	dv.LayoutDirection = PlatformView.LayoutDirection;
}

var lv = _dialog?.ListView;
if (lv != null)
{
	lv.LayoutDirection = PlatformView.LayoutDirection;
}
```

right after `_dialog = builder.Create();`

Reply:
Yes, this was exactly my first approach https://github.com/dotnet/maui/pull/13672/commits/02c7d609d3ef3bd3df938d6a9ad1f4af4af93263
Then, added the extension methods to allow to reuse anywhere else in case it is necessary. Let me move back to the initial approach.